view/src/service.rs

```diff
@@ -187,6 +189,38 @@ impl ViewProtocol for ViewService {
         self.check_worker().await?;
         self.check_fvk(request.get_ref().fvk_hash.as_ref()).await?;
 
-        todo!()
+        // Acquire a read lock for the NCT that will live for the entire request,
+        // so that all auth paths are relative to the same NCT root.
+        let nct = self.note_commitment_tree.read().await;
+
+        // Read the NCT root
+        let anchor = nct.root();
+
+        // Obtain an auth path for each requested note commitment
+        let requested_note_commitments = request
+            .get_ref()
+            .note_commitments
+            .iter()
+            .map(|nc| {
```

Comment:
We shouldn't expect here, since that will bring down the whole server on a malformed request.

Instead, if you do .map(Commitment::try_from), you'll get an iterator of Result s, and if you do .collect::<Result<Vec<_>, _>>()? you can collect the iterator of Results into a Result of a Vec, then bubble up any conversion errors.

Reply:
I was wondering about that. FYI we `expect` on protobuf conversions from request values elsewhere in the service (e.g. https://github.com/penumbra-zone/penumbra/blob/main/view/src/service.rs#L160), I can fix those up as well?

Reply:
Yeah, that sounds good!